salt/returners/cassandra_cql_return.py

```diff
@@ -0,0 +1,390 @@
+                 '{0}', '{1}'
+               );'''.format(ret['id'], ret['fun'])
+
+    # cassandra_cql.cql_query may raise a CommandExecutionError
+    try:
+        __salt__['cassandra_cql.cql_query'](query)
+    except CommandExecutionError:
+        log.critical('Could not store minion ID with Cassandra returner.')
+        raise
+    except Exception as e:
+        log.critical('Unexpected error while inserting minion ID into the minions table: {0}'.format(str(e)))
+        raise
+
+
+def event_return(events):
+    '''
+    Return event to one of potentially many clustered cassandra nodes
+
+    Requires that configuration be enabled via 'event_return'
+    option in master config.
+
+    Cassandra does not support an auto-increment feature due to the
+    highly inefficient nature of creating a monotonically increasing
+    number accross all nodes in a distributed database. Each event
+    will be assigned a uuid by the connecting client.
+    '''
+    for event in events:
+        tag = event.get('tag', '')
+        data = event.get('data', '')
+        query = '''INSERT INTO salt.salt_events (
+                     id, alter_time, data, master_id, tag
+                   ) VALUES (
+                     {0}, {1}, '{2}', '{3}', '{4}'
+                   );'''.format(str(uuid.uuid1()),
+                                int(time.time() * 1000),
+                                json.dumps(data), 
+                                __opts__['id'], 
+                                tag)
+
+        # cassandra_cql.cql_query may raise a CommandExecutionError
+        try:
+            __salt__['cassandra_cql.cql_query'](query)
+        except CommandExecutionError:
+            log.critical('Could not store events with Cassandra returner.')
+            raise
+        except Exception as e:
+            log.critical('Unexpected error while inserting into salt_events: {0}'.format(str(e)))
+            raise
+
+
+def save_load(jid, load):
+    '''
+    Save the load to the specified jid id
+    '''
+    query = '''INSERT INTO salt.jids (
+                 jid, load
+               ) VALUES (
+                 '{0}', '{1}'
+               );'''.format(jid, json.dumps(load))
+
+    # cassandra_cql.cql_query may raise a CommandExecutionError
+    try:
+        __salt__['cassandra_cql.cql_query'](query)
+    except CommandExecutionError:
+        log.critical('Could not save load in jids table.')
+        raise
+    except Exception as e:
+        log.critical('Unexpected error while inserting into jids: {0}'.format(str(e)))
+        raise
+
+
+# salt-run jobs.list_jobs FAILED
+def get_load(jid):
+    '''
+    Return the load data that marks a specified jid
+    '''
+    query = '''SELECT load FROM salt.jids WHERE jid = '{0}';'''.format(jid)
+
+    ret = {}
+
+    # cassandra_cql.cql_query may raise a CommandExecutionError
+    try:
+        data = __salt__['cassandra_cql.cql_query'](query)
+        if data:
+            load = data[0].get('load')
```

Comment:
This might need more error checking. Could see other exceptions floated from module and ```data``` could be a zero-item list.